storage/shard.go

```diff
@@ -893,6 +881,16 @@ func (s *dbShard) FlushState(blockStart time.Time) fileOpState {
 	return state
 }
 
+func (s *dbShard) markFlushStateSuccessOrError(blockStart time.Time, err error) error {
```

Comment:
Yeah I guess I don't feel that's idiomatic Go, even though I'm a fan of that in other languages (although more the fluid interface of returning instance to self/this rather than an input argument).